test_scripts/Capabilities/PersistingHMICapabilities/common.lua

```diff
@@ -0,0 +1,419 @@
+end
+
+function m.registerApp(pAppId, pCapResponse, pMobConnId, hasPTU)
+  if not pCapResponse then pCapResponse = {} end
+  if not pAppId then pAppId = 1 end
+  if not pMobConnId then pMobConnId = 1 end
+
+  local policyModes = {
+    P  = "PROPRIETARY",
+    EP = "EXTERNAL_PROPRIETARY",
+    H  = "HTTP"
+  }
+
+  local session = m.createSession(pAppId, pMobConnId)
+  session:StartService(7)
+  :Do(function()
+     local msg = ""
+      local corId = session:SendRPC("RegisterAppInterface", m.getParams(pAppId))
+      m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered",
+        { application = { appName = m.getParams(pAppId).appName } })
+      :Do(function(_, d1)
+          m.setHMIId(d1.params.application.appID, pAppId)
+          if hasPTU then
+            m.ptu.expectStart()
+          end
+        end)
+      session:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" }  )
+      :ValidIf(function(_,data)
+        for key, value in pairs (pCapResponse) do
+          print(key)
+          if not m.isTableEqual(data.payload[key], pCapResponse[key]) then
+            msg = msg .. key.." contains unexpected parameters\n"..
+              " Expected: " .. m.tableToString(pCapResponse[key]) .. "\n" ..
+              " Actual: " .. m.tableToString(data.payload[key]) .. "\n"
+          end
+        end
+        if string.len(msg) > 0 then
+          return false, msg
+        else
+          return true
+        end
+      end)
+      :Do(function()
+          session:ExpectNotification("OnHMIStatus",
+            { hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+          session:ExpectNotification("OnPermissionsChange")
+          :Times(AnyNumber())
+          local policyMode = SDL.buildOptions.extendedPolicy
+          if policyMode == policyModes.P or policyMode == policyModes.EP then
+            session:ExpectNotification("OnSystemRequest", { requestType = "LOCK_SCREEN_ICON_URL" })
+          end
+        end)
+    end)
+end
+
+function m.masterReset(pExpFunc)
+  local isOnSDLCloseSent = false
+  if pExpFunc then pExpFunc() end
+  m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "MASTER_RESET" })
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLClose")
+  :Do(function()
+    isOnSDLCloseSent = true
+    SDL.DeleteFile()
+  end)
+  :Times(AtMost(1))
+  m.wait(3000)
+  :Do(function()
+    if isOnSDLCloseSent == false then utils.cprint(35, "BC.OnSDLClose was not sent") end
+    if SDL:CheckStatusSDL() == SDL.RUNNING then SDL:StopSDL() end
+  end)
+end
+
+function m.ignitionOff()
+  config.ExitOnCrash = false
+  local timeout = 5000
+  local function removeSessions()
+    for i = 1, m.getAppsCount() do
+      test.mobileSession[i] = nil
+    end
+  end
+  local event = events.Event()
+  event.matches = function(event1, event2) return event1 == event2 end
+  EXPECT_EVENT(event, "SDL shutdown")
```

Comment:
@VjKlepikov Please use `actions.mobile.getConnection():ExpectEvent ` instead of  `EXPECT_EVENT`

Reply:
done